font/src/ft/mod.rs

```diff
@@ -293,19 +283,24 @@ impl FreeTypeRasterizer {
         };
 
         // Reuse the font_key, since changing it can break library users
-        let font_key = self.face_from_pattern(&base_font, font_key).and_then(|pattern| {
-            pattern.map(Ok).unwrap_or_else(|| Err(Error::MissingFont(desc.to_owned())))
-        })?;
+        let font_key = self.face_from_pattern(&primary_font, primary_font_id, font_key).and_then(
+            |pattern| pattern.map(Ok).unwrap_or_else(|| Err(Error::MissingFont(desc.to_owned()))),
+        )?;
 
         // Coverage for fallback fonts
         let coverage = CharSet::new();
         let empty_charset = CharSet::new();
-        // Load fallback list
-        let list: Vec<FallbackFont> = font_iter
-            .map(|font| {
-                let charset = font.get_charset().unwrap_or(&empty_charset);
+
+        // Build fallback list
+        let list: Vec<FallbackFont> = matched_fonts
+            .map(|fallback_pattern| {
+                let charset = fallback_pattern.get_charset().unwrap_or(&empty_charset);
+                let fallback_font = primary_font.render_prepare(config, fallback_pattern);
```

Comment:
I don't think we should name patterns `font`. If we name it pattern, everyone immediately knows what we're talking about.